<launch>
<!--
  <node name="yaw_pid" pkg="pid" type="controller" output="screen" >
    <param name="Kp" value="-0.4" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.0" />
    <param name="upper_limit" value="100" />
    <param name="lower_limit" value="-100" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="10" />
    <param name="diagnostic_period" value="0.25" />
    <param name="max_loop_frequency" value="50.0" />
    <param name="min_loop_frequency" value="0.0" />
  </node>
 -->

  <node name="controller" pkg="pid" type="controller" ns="x_controller" output="screen" >
    <param name="node_name" value="x_pid" />
    <param name="Kp" value="1.0" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.0" />
    <param name="upper_limit" value="10" />
    <param name="lower_limit" value="-10" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="5" />
    <param name="diagnostic_period" value="0.25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="0.0" />
    <remap from="setpoint" to="/x_controller/setpoint" />
  </node>


  <node name="controller" pkg="pid" type="controller" ns="y_controller" output="screen" >
    <param name="node_name" value="y_pid" />
    <param name="Kp" value="1.0" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.0" />
    <param name="upper_limit" value="10" />
    <param name="lower_limit" value="-10" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="5" />
    <param name="diagnostic_period" value="0.25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="0.0" />
    <remap from="setpoint" to="/y_controller/setpoint" />
  </node>


  <node name = "pose_estimation" pkg = "ardrone_follow_tag" type = "pose_estimation.py"/>
  <node name = "yaw_controller" pkg = "ardrone_follow_tag" type = "test_yaw_controller.py"/>

  <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/x_controller/control_effort/data /x_controller/state/data /y_controller/control_effort/data y_controller/state/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
    <!-- <node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" /> -->
  </launch>
